qiskit_machine_learning/algorithms/classifiers/neural_network_classifier.py

```diff
@@ -66,7 +66,7 @@ def __init__(
                 as a set of individual predictions with occurrence probabilities (the index would be
                 the prediction and the value the corresponding frequency, e.g. for absolute/squared
                 loss). This option is ignored in case of a one-dimensional output.
-            optimizer: An instance of an optimizer to be used in training.
+            optimizer: An instance of an optimizer to be used in training. Default is SLSQP.
```

Comment:
Can we make the default behavior a little clearer for a reader/user by saying something like 
```
When `None` defaults to SLSQP.
```
instead of just `Default is SLSQP`.

Same for the other classes

Reply:
Should the reno be changed too?

Reply:
I think the reno is fine - its says we now have SLSQP as a default. Here, where its closer to the code, where the user needs to be a bit more aware of how the default comes about, i,e, via None, I think the docstrings suffice so they can find how how if/when they need to.